File: Core.DomainModel/ItSystem/ItSystemRight.cs
namespace Core.DomainModel.ItSystem
{
    /// <summary>
    /// Associates a <see cref="User"/> with an it system (<see cref="Object"/>) in a specific <see cref="Role"/>.
    /// </summary>
    public class ItSystemRight : Entity, IRight<ItSystemUsage.ItSystemUsage, ItSystemRight, ItSystemRole>
    {
        public int UserId { get; set; }
        public int RoleId { get; set; }
        public int ObjectId { get; set; }
        public virtual User User { get; set; }
        public virtual ItSystemRole Role { get; set; }
        public virtual ItSystemUsage.ItSystemUsage Object { get; set; }
    }
}

File: Presentation.Web/app/models/it-system/it-system-right.ts
module Kitos.Models.ItSystem {
    export interface IItSystemRight {
        UserId: number;
        RoleId: number;
        ObjectId: number;
        User: IUser;
        Role: IRoleEntity;
        Object: ItSystemUsage.IItSystemUsage;
    }
}
